launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -455,6 +542,36 @@ private ProcessBuilder createBuilder() {
     for (Map.Entry<String, String> e : builder.childEnv.entrySet()) {
       pb.environment().put(e.getKey(), e.getValue());
     }
+
+    if (builder.workingDir != null) {
+      pb.directory(builder.workingDir);
+    }
+
+    // Only one of redirectError, redirectError(...) can be specified.
+    // If redirectToLog() is specified, none of the other redirect options can be specified
+    if (builder.redirectToLog &&
```

Comment:
I think this should follow the semantics of `ProcessBuilder`, which is "last one wins". Try this out:

```
 new ProcessBuilder("ls").redirectOutput(ProcessBuilder.Redirect.PIPE).redirectOutput(ProcessBuilder.Redirect.INHERIT).start()
```